src/CommentSold/Api/Services/AccountService.php

```diff
@@ -9,49 +9,52 @@
 use CommentSold\Api\Resources\Request\Account\CreateShopRequest;
 use CommentSold\Api\Resources\Request\Account\GetIframeUrlRequest;
 use CommentSold\Api\Resources\Request\Account\GetOauthUrlRequest;
+use CommentSold\Api\Resources\Response\Account\CreateShopResponse;
+use CommentSold\Api\Resources\Response\Account\GetIframeUrlResponse;
+use CommentSold\Api\Resources\Response\Account\GetOauthUrlResponse;
 use CommentSold\Api\ShopClient;
 
 class AccountService extends abstractService
 {
     /**
      * Create Shop
      */
-    public function createShop(CreateShopRequest $payload)
+    public function createShop(CreateShopRequest $payload): CreateShopResponse
     {
         if (! $this->client instanceof GlobalClient) {
             throw new InvalidContextException('Global client required');
         }
 
         $response = $this->restClient->post('accounts', $payload);
 
-        return $response->toObject();
+        return new CreateShopResponse($response);
```

Comment:
Is there technical difficulty that prevents this returning `Customer`? 

In general, if you have a VO representing a CS resource, that VO can be returning from the various type of endpt operating on that resource. I'm not seeing what the "type-response" achieve since to access the customer data, you gotta do `->toArray()`.

Reply:
The idea here was that if I return a Response then the SDK user could choose to use `->toArray` or `->toJson` (or anything else we add later) and we could also later pack additional methods to gain insight into the responses. 

Perhaps we add a helper method like `->getRateLimit()` which is a helper to return some information that's in the header.

Reply:
> The idea here was that if I return a Response then the SDK user could choose to use `->toArray` or `->toJson` (or anything else we add later) and we could also later pack additional methods to gain insight into the responses.

You have `Customer` wrapping your payload in `GetCustomersResponse`. Why just there and not elsewhere?  Note that you got three forms of "customer" (VO, array, json) available.

Reply:
`GetCustomersResponse` (or any of the paginated responses) I need to split the pagination from the data. So there is `data` and `pagination`.

But with the example I gave above, and a little more time thinking about it, maybe I update all the responses to have the data in `data` even if it doesn't make sense to do it (like the entire response is data) but have `->getData()`, `->getPagination()`, etc. helpers that you would use. And then you can still do `->getJson()` or `->getArray()` on that.

So it would be like this:
```
$response = $client->getCustomers();

$totalCustomers = $response->getPagination()->totalCount;
$customers = $response->getData()->toArray(); // if you wanted as an array. Otherwise defaults to object
```

I know it's an extra thing but it gives us a lot of flexibility.

Reply:
Sure, that could work.